hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -177,7 +178,8 @@ private boolean hasNotExpired(QueryableEntry queryableEntry) {
         }
 
         RecordStore recordStore = mapServiceContext.getExistingRecordStore(partitionId, name);
-        return recordStore != null && !recordStore.expireOrAccess(keyData);
+        return recordStore != null
+                && !recordStore.isExpired(keyData, Clock.currentTimeMillis(), false);
```

Comment:
`Clock.currentTimeMillis()` might be a pretty expensive call in some environments, it might be better to obtain the stamp just once outside of the scan loop.

Reply:
fixed